src/ripple/basics/SlabAllocator.h

```diff
@@ -100,91 +126,109 @@ class SlabAllocator
         SlabBlock(SlabBlock&& other) = delete;
         SlabBlock&
         operator=(SlabBlock&& other) = delete;
+    };
 
-        /** Determines whether the given pointer belongs to this allocator */
-        bool
-        own(std::uint8_t const* p) const noexcept
-        {
-            return (p >= p_) && (p < p_ + size_);
-        }
+private:
+    // A linked list of slabs
+    std::atomic<SlabBlock*> slabs_ = nullptr;
 
-        std::uint8_t*
-        allocate() noexcept
-        {
-            std::uint8_t* ret;
+    // A pointer to the slab we're attempting to allocate from
+    std::atomic<SlabBlock*> active_ = nullptr;
 
-            {
-                std::lock_guard l(m_);
+    // The size of an item, including the extra bytes requested and
+    // any padding needed for alignment purposes between items:
+    std::size_t const itemSize_;
 
-                ret = l_;
+    // The maximum number of items that this slab can allocate
+    std::size_t const itemCount_;
 
-                if (ret)
-                {
-                    // Use memcpy to avoid unaligned UB
-                    // (will optimize to equivalent code)
-                    std::memcpy(&l_, ret, sizeof(std::uint8_t*));
-                }
-            }
+    // The size of each individual slab:
+    std::size_t const slabSize_;
 
-            return ret;
+private:
+    std::uint8_t*
+    allocate(SlabBlock* slab) noexcept
+    {
+        std::uint8_t* ret = nullptr;
+
+        if (slab->data_ != nullptr)
```

Comment:
I believe this is the only place where slab `data_` is accessed outside of a lock, and it's to check for what will be a rare occurrence and then immediately takes out a lock. What do you think about putting this inside a lock and removing all the atomic variables/atomic swaps and making the mutex protect `data_`?